helper/pipelinehelper/pipelinehelper.go

```diff
@@ -12,4 +14,10 @@ const (
 // GetRealPipelineName removes the suffix from the pipeline name.
 func GetRealPipelineName(name string, pType gaia.PipelineType) string {
 	return strings.TrimSuffix(name, typeDelimiter+pType.String())
-}
\ No newline at end of file
+}
+
+// AppendTypeToName appends the type to the output binary name.
+// This allows later to define the pipeline type by the pipeline binary name.
+func AppendTypeToName(n string, pType gaia.PipelineType) string {
```

Comment:
Can we try eliminating these helpers please or put them where they belong? The helper/stuff/util packages are a architectural clutch/anti-pattern. It's code of which no-one knows where to put or was put there for rudimentary decoupling. I'm sure we can find a place for these to live. Maybe their own package something that makes sense at the first sight? Like I look at helpers and I will have no idea what code to expect in there. Where as if I would look at package pipeline I'll expect things relating to pipelines.

Reply:
I personally like having a `helper` package and I disagree that this is an architectural clutch/anti-pattern. Often, you find yourself in a situation where you have a few small and simple functions which you need to use in several different packages. If you put them in the `pipeline` package, you face quickly circular dependency issues which is a problem.
We could move all packages inside the helper package to the top-level but this makes the package structure way messier. 

> I look at helpers and I will have no idea what code to expect in there

The package structure is usually helper/<package-name> (e.g. helper/stringhelper). I personally think this is self-explaining. 

> Where as if I would look at package pipeline I'll expect things relating to pipelines.

So you are not confused when the package is called `pipelines` but you are when it is called `pipelinehelper`? 😄 

I also made some good experiences with this structure in other big projects (Vault: https://github.com/hashicorp/vault/tree/master/helper).

Reply:
Anyways. At least it's covered with tests.

Reply:
> The stringhelper is a good example that that thing could easily live in its own package. In fact, you already did put it into it's own package. :D

That is my point. It is already in its own package. I don't want to move that package to the top-level. Imagine you are new to the project and you need to find out where everything exists. If we have 20+ folders in the top-level folder, it will be a huge mess. The helper package provides a little benefit which makes the project structure a bit cleaner. 

> Yeah I know other big projects use helper, and I will strongly advocate in those big projects against it as well. :D

I know there are some Go-veterans on the internet which have the same opinion. I totally understand that point but I found out that it is simply not practicable. 😄

Reply:
Hm. I guess I can see that in this case it's okay, since it's in its own package that makes sense. I gues the problem is when the helper is just a jumble of stuff extracted for the sake of extraction. What I don't want to see is code that has no purpose or exists in a utils package we don't know what it does or how it's used and the extraction only happened because we wanted to avoid to copy a bit of code, or to avoid loops in imports. :)

What I'm saying is that I can live with this. :DDDDD